plugins/patchmanager.koplugin/main.lua

```diff
@@ -0,0 +1,170 @@
+    name = "patchmanager",
+    is_doc_only = false,
+    patches = {}
+}
+
+function PatchManager:init()
+    self.patch_dir = DataStorage:getDataDir() .. "/patches"
+    self.disable_ext = ".disabled"
+    self:getAvailablePatches()
+    self.ui.menu:registerToMainMenu(self)
+end
+
+function PatchManager:getAvailablePatches()
+    if lfs.attributes(self.patch_dir, "mode") ~= "directory" then
+        self.patches = {}
+        return
+    end
+
+    for priority = tonumber(UserPatch.early_once), tonumber(UserPatch.on_exit) do
+        self.patches[priority] = {}
+        for entry in lfs.dir(self.patch_dir) do
+            local mode = lfs.attributes(self.patch_dir .. "/" .. entry, "mode")
+            if entry and mode == "file" and entry:match("^" .. priority .. "%d*%-") then
+                if entry:find(".lua", 1, true) then
+                    -- exclude any non lua files
+                    table.insert(self.patches[priority], entry)
+                end
+            end
+        end
+    end
+end
+
+function PatchManager:getSubMenu(priority)
+    if priority < tonumber(UserPatch.early_once) or priority > tonumber(UserPatch.on_exit) then
+        logger.err("PatchManager: BUG, wrong patch_level:", priority)
+    end
+    local sub_menu = {}
+    if #self.patches == 0 then
+        return {}
+    end
+    for i = 1, #self.patches[priority] do
+        local patch_name = self.patches[priority][i]:sub(1, self.patches[priority][i]:find(".lua", 1, true) + 3)
+        table.insert(sub_menu, {
+            text = patch_name,
+            checked_func = function()
+                return self.patches[priority][i]:find("%.lua$") ~= nil
+            end,
+            callback = function()
+                local is_patch_enabled = self.patches[priority][i]:find("%.lua$") ~= nil
+                if is_patch_enabled then
+                    local disabled_name = self.patches[priority][i] .. self.disable_ext
+                    os.remove(self.patch_dir .. "/" .. disabled_name) -- remove a possible leftover (caused by user)
+                    os.rename(self.patch_dir .. "/" .. self.patches[priority][i],
+                              self.patch_dir .. "/" .. disabled_name)
+                    self.patches[priority][i] = disabled_name
+                else
+                    local pos = self.patches[priority][i]:find(".lua", 1, true)
+                    if pos then
+                        local enabled_name = self.patches[priority][i]:sub(1, pos + (".lua"):len() - 1)
+                        os.rename(self.patch_dir .. "/" .. self.patches[priority][i],
+                                  self.patch_dir .. "/" .. enabled_name)
+                        self.patches[priority][i] = enabled_name
+                    end
+                end
+                UIManager:askForRestart(_("Patches changed. Current patchset will be applied on next restart."))
+            end,
+            hold_callback = function()
```

Comment:
Maybe call TextEditor?

Reply:
There is a typo: patches spelt as patchse

Reply:
I think it is really meant as "patch set" (dunno if this can/should be made a single word "patchset" in correct English :)

Reply:
I think it's correct but programmer jargon. Set of patches might suffice to make it less so.